integration/src/main/java/ai/djl/integration/tests/training/listener/EarlyStoppingListenerTest.java

```diff
@@ -0,0 +1,116 @@
+package ai.djl.integration.tests.training.listener;
+
+import ai.djl.Model;
+import ai.djl.basicdataset.cv.classification.Mnist;
+import ai.djl.basicmodelzoo.basic.Mlp;
+import ai.djl.integration.util.TestUtils;
+import ai.djl.metric.Metrics;
+import ai.djl.ndarray.types.Shape;
+import ai.djl.nn.Activation;
+import ai.djl.training.DefaultTrainingConfig;
+import ai.djl.training.EasyTrain;
+import ai.djl.training.Trainer;
+import ai.djl.training.TrainingResult;
+import ai.djl.training.dataset.Dataset;
+import ai.djl.training.listener.EarlyStoppingListener;
+import ai.djl.training.listener.TrainingListener;
+import ai.djl.training.loss.Loss;
+import ai.djl.training.optimizer.Optimizer;
+import ai.djl.training.tracker.Tracker;
+import ai.djl.translate.TranslateException;
+import org.testng.Assert;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+
+public class EarlyStoppingListenerTest {
+
+    private final Optimizer sgd = Optimizer.sgd().setLearningRateTracker(Tracker.fixed(0.1f)).build();
+
+    private Mnist testMnistDataset;
+    private Mnist trainMnistDataset;
+
+    @BeforeTest
+    public void setUp() throws IOException, TranslateException {
+        testMnistDataset = Mnist.builder()
```

Comment:
You can set the `limit` on the dataset to make it act like it only has a smaller size. This will let these tests be sped up any further. We try to keep most of our unit tests as quick as possible because that makes our build and CI run faster and we have to spend less time waiting for it

Reply:
Set the limit, and reduced the run time to 700ms (was 6 seconds) and now has a extra test as well